volume-group-operator/controllers/utils/volumegroup.go

```diff
@@ -0,0 +1,46 @@
+package utils
+
+import (
+	volumegroupv1 "github.com/IBM/volume-group-operator/api/v1"
+	"github.com/go-logr/logr"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func UpdateVolumeGroupSource(instance *volumegroupv1.VolumeGroup, vgc *volumegroupv1.VolumeGroupContent) {
+	instance.Spec.Source = volumegroupv1.VolumeGroupSource{
+		VolumeGroupContentName: &vgc.Name,
+		Selector:               instance.Spec.Source.Selector,
+	}
+}
+
+func updateVolumeGroupStatus(client client.Client, instance *volumegroupv1.VolumeGroup, logger logr.Logger) error {
+	if err := UpdateObjectStatus(client, instance); err != nil {
+		logger.Error(err, "failed to update status")
+
+		return err
+	}
+	return nil
+}
+
+func UpdateVolumeGroupStatus(client client.Client, instance *volumegroupv1.VolumeGroup, vgc *volumegroupv1.VolumeGroupContent,
+	groupCreationTime *metav1.Time, ready bool, logger logr.Logger) error {
+	instance.Status = volumegroupv1.VolumeGroupStatus{
```

Comment:
can you add all the Status fields? because it overwrites fields that do not appear here, can you also do it for the `volumeGroupContent` and class if needed?

Reply:
done

Reply:
what about the PVCList and PVList fields in the `volumeGrouo` and `volumeGroupContent`?

Reply:
I changed it so no we did not override the Status but only change the fields we need on the existing one